runtimes/weave/src/lib.rs

```diff
@@ -2534,6 +2537,12 @@ impl_runtime_apis! {
 			vec![]
 		}
 	}
+
+	impl crate::registry_entry_api::RegistryEntryApi<Block, Hash, RegistryIdOf> for Runtime {
```

Comment:
Can't we use the Ss58Identifier type here instead of RegistryIdOf. Good to keep runtime out of pallet-derived types.